### DISCUSSION TITLE
Generic IQueryable projections

### DISCUSSION THREAD

**Discussion by 19bartek92** [riok/mapperly#960] [Q&A]
How to handle generic IQueryable projections? Is it possible to do this? 
I'm looking for the equivalent of the "ProjectTo" extension from AutoMapper to add to my generic repository.

**Comment by TimothyMakkison** [comment]
Hey, Mapperly supports [derived type](https://mapperly.riok.app/docs/configuration/derived-type-mapping/) mapping for IQueryable projections.
```C#
public static partial System.Linq.IQueryable<B> Map(System.Linq.IQueryable<A> source);

[MapDerivedType<ASubType1, BSubType1>]
[MapDerivedType<ASubType2, BSubType2>]
public static partial B Map(A src);
```

I don't think generic mappings are supported yet. I'll see if I can add it 👍

**Comment by latonz** [comment]
Created https://github.com/riok/mapperly/issues/968 to track this.
